File: OOP September 2014/Homeworks/06_Encapsulation-and-Polymorphism/01_Shapes/Circle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _01_Shapes
{
    public class Circle : BasicShape, IShape
    {
        private double radius;

        public double Radius 
        {
            get { return this.radius;} 
            set
            {
                if (value < 0)
                {
                    throw new ArgumentOutOfRangeException("Radius cannot be negative.");
                }
                this.radius = value;
            }
        }

        public Circle(double radius)
            : base(radius, radius)
        {
            this.Radius = radius;
        }

        public override double CalculateArea()
        {
            return Math.PI * this.Radius * this.Radius;
        }

        public override double CalculatePerimeter()
        {
            return 2 * Math.PI * this.Radius;
        }
    }
}

File: OOP September 2014/Lectures/08_OOP-in-Java/8. OOP-in-Java-Demos/src/shapes/Circle.java
package shapes;

public class Circle extends Shape {
	private double radius;
	
	public Circle(double x, double y, double radius) {
		super(x, y);
		this.radius = radius;
	}

	public double getRadius() {
		return radius;
	}

	public void setRadius(double radius) {
		this.radius = radius;
	}

	@Override
	public double calculateArea() {
		return Math.PI * this.radius * this.radius;
	}
	
	@Override
	public String toString() {
		return "Circle(" + this.x + ", " + this.y + 
				", r=" + this.radius + ")";
	}
}
